src/Cuda.cmake

```diff
@@ -0,0 +1,51 @@
+# ! find_and_link_cuda
+# A function that links Cuda to the given target. This function automatically links the compiler warnings, so do not link `project_warnings` with your target.
+#
+# # Example
+# add_executable(main_cuda main.cu)
+# target_compile_features(main_cuda PRIVATE cxx_std_17)
+# target_link_libraries(main_cuda PRIVATE project_options)
+# find_and_link_cuda(main_cuda)
+#
+macro(find_and_link_cuda target)
+  # optional named CUDA_WARNINGS
+  set(oneValueArgs CUDA_WARNINGS)
+  cmake_parse_arguments(
+    _cuda_args
+    ""
+    "${oneValueArgs}"
+    ""
+    ${ARGN})
+
+  # add CUDA to cmake language
+  enable_language(CUDA)
+
+  # use the same C++ standard if not specified
+  if("${CMAKE_CUDA_STANDARD}" STREQUAL "")
+    set(CMAKE_CUDA_STANDARD "${CMAKE_CXX_STANDARD}")
+  endif()
+
+  # -fPIC
+  set_target_properties(${target} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+  # We need to explicitly state that we need all CUDA files in the
+  # ${target} library to be built with -dc as the member functions
+  # could be called by other libraries and executables
+  set_target_properties(${target} PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+
+  if(APPLE)
+    # We need to add the path to the driver (libcuda.dylib) as an rpath,
+    # so that the static cuda runtime can find it at runtime.
+    set_property(TARGET ${target} PROPERTY BUILD_RPATH ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
+  endif()
+
+  if(${_cuda_args_CUDA_WARNINGS} STREQUAL "")
+    target_compile_options(
+      ${target}
+      PRIVATE -Wall
+              -Wextra
+              -Wunused
+              -Wconversion
+              -Wshadow)
+  endif()
```

Comment:
I think we should remove this, and then add the Cuda warnings in `CompilerWarnings.cmake` like how @DStrelak suggested in https://github.com/cpp-best-practices/project_options/issues/47#issuecomment-1024343401

Reply:
How does your suggestion manage mixed languages? Does that add warnings for both CXX and Cuda?